R/cchart.R

```diff
@@ -299,6 +302,34 @@ CChart <- function(chart.type, x, small.multiples = FALSE,
     result
 }
 
+addWarning <- function(x, chart.type, small.multiples, has.annotations)
+{
+    export.type <- attr(x, "ChartType")
+    msg <- ""
+
+    if (small.multiples)
+        msg <- "This visualization is a small multiple which is not supported by Microsoft."
+    else if (has.annotations)
+        msg <- "This visualization contains annotations which is not supported by Microsoft."
+    else if (chart.type %in% c("Palm", "Stream", "Venn", "Pyramid",
+            "BarPictograph", "StackedColumnWithStatisticalSignificance"))
+        msg <- "This visualization type is not supported by Microsoft."
+    else if (export.type %in% c("Sunburst", "Histogram", "Filled Map",
+            "Box & Whisker"))
+        msg <- "This visualization type cannot be exported to PowerPoint."
```

Comment:
For these viz types, could the name be inserted into the `msg`? e.g.
```
msg <- paste("The", export.type, "visualization cannot be exported to PowerPoint")
```

Reply:
Could also update the message to say that the reason is the chart type doesn't exist in PowerPoint (if true of course). That way the user knows that the chart cant be made in PowerPoint at all, not that the viz cant be translated to PowerPoint. i.e. its a limitation of Microsoft, not Displayr.

Reply:
The warnings in line 319 is actually because the chart type exists in PowerPoint, but Displayr's API doesn't support their export. The reason why I haven't specified what the export type is, is that for user this export type seems to come from nowhere. For example, if the chart type is "Geographic Map", the export type is "Filled Map" (which is the name of the relevant chart type in Powerpoint), but Filled Map is not supported